cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -280,6 +280,7 @@ func main() {
 
 	// Generate a unique ID for this provisioner
 	timeStamp := time.Now().UnixNano() / int64(time.Millisecond)
+	rand.Seed(timeStamp)
```

Comment:
I agree that `rand.Intn` wasn't useful. But is this any better?

The randomness of `identity` still depends entirely on `timeStamp` - if we can't improve that, then we might as well drop the `rand.Intn` part.

Does Go have an API for the system's random number pool?

Reply:
The standard library has `math/rand`, which is not cryptographically secure, and `crypto/rand`, which should just read from `/dev/random` on Linux. The easiest option would be to read 2-4 bytes of dev random, hex encode them and use that instead.

Finding another seed value for math random is another option, but the most likely choices I can see are time-based and the driver name, which are already there in that string.

Reply:
... And process id is another classic option, which makes the ID string unlikely to collide combined with the time-based int already in the string.

Reply:
I like using `crypto/rand`.

Reply:
+1 for `crypto/rand`, I have mostly seen that used to solve this issue with `math/rand`